cocotb/handle.py

```diff
@@ -778,7 +783,12 @@ def _set_value(self, value, call_sim):
                 "Unsupported type for integer value assignment: {} ({!r})"
                 .format(type(value), value))
 
-        call_sim(self._handle.set_signal_val_long, set_action, value)
+        if value < _INT_MIN or value > _INT_MAX:
```

Comment:
Tempting to just handle this in C, where we actually require this to be true.

Reply:
Will do